spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8SecretsPropertySource.java

```diff
@@ -49,22 +50,27 @@ public Fabric8SecretsPropertySource(KubernetesClient client, String name, String
 			Map<String, String> labels, boolean failFast) {
 		Map<String, Object> result = new HashMap<>();
 
-		LOG.info("Loading Secret with name '" + name + "' or with labels [" + labels + "] in namespace '" + namespace
-				+ "'");
 		try {
 
-			Secret secret = client.secrets().inNamespace(namespace).withName(name).get();
-
-			// the API is documented that it might return null
-			if (secret == null) {
-				LOG.warn("secret with name : " + name + " in namespace : " + namespace + " not found");
-			}
-			else {
-				putDataFromSecret(secret, result, namespace);
+			// can come as null or empty, totally valid scenario
+			if (StringUtils.hasText(name)) {
```

Comment:
I think I would prefer we switch the logic here

if there are labels ignore the name and load the secrets that match the label
if there are no label check if there is a name (there always should be but worth a check) and load that if present

Then you could pass the secret name and it would still work as expected.

Reply:
I was actually torn apart, to either do what you suggest here, or what I did.

The problem with your suggestion is that it is a breaking change, someone that configured this with both a name and labels will get un-expected results, right?

Reply:
Can we have two separate constructors, one for name based and one for labels based?

Reply:
If you are thinking to have that _in general_, then yeah, why not. If you are thinking to have that here, in this PR, then I'd say no. This will complicate things a lot as this was will require many changes. Especially if we want to implement [this filters](https://github.com/spring-cloud/spring-cloud-kubernetes/pull/818), the the idea of "sources" will become less relevant, as searching for things will be done in a different way.